torch/nn/modules/module.py

```diff
@@ -1954,7 +1959,15 @@ def _load_from_state_dict(self, state_dict, prefix, local_metadata, strict,
                     continue
                 try:
                     with torch.no_grad():
-                        param.copy_(input_param)
+                        if assign_to_params_buffers:
+                            # Shape checks are already done above
+                            is_persistent_buffer = name in persistent_buffers
+                            if is_persistent_buffer:
```

Comment:
Why is this based on persistent buffer and not `if isinstance(param, torch.nn.Parameter)` ?